lib/pluggy_elixir/transaction.ex

```diff
@@ -0,0 +1,214 @@
+       %{
+         page: 1,
+         total: 1,
+         total_pages: 1,
+         transactions: [
+           %Transaction{
+             account_id: "03cc0eff-4ec5-495c-adb3-1ef9611624fc",
+             amount: 1_500,
+             balance: 3_000,
+             category: "Transfer",
+             currency_code: "BRL",
+             date: ~N[2021-04-12 00:00:00.000],
+             description: "TED Example",
+             description_raw: nil,
+             id: "6ec156fe-e8ac-4d9a-a4b3-7770529ab01c",
+             payment_data: %{
+               payer: %{
+                 account_number: "1234-5",
+                 branch_number: "090",
+                 document_number: %{
+                   type: "CPF",
+                   value: "882.937.076-23"
+                 },
+                 routing_number: "001",
+                 type: nil
+               },
+               payment_method: "TED",
+               reason: "Taxa de serviço",
+               receiver: %{
+                 account_number: "9876-1",
+                 branch_number: "999",
+                 document_number: %{
+                   type: "CNPJ",
+                   value: "08.050.608/0001-32"
+                 },
+                 routing_number: "002",
+                 type: nil
+               },
+               reference_number: "123456789"
+             },
+             provider_code: "123",
+             status: "POSTED"
+           }
+         ]
+       }}
+  """
+
+  @spec all_by_account(
+          %{
+            :account_id => String.t(),
+            :from => Date.t(),
+            :to => Date.t(),
+            optional(:page_size) => integer(),
+            optional(:page) => integer()
+          },
+          Config.config_overrides()
+        ) ::
+          {:ok, %{page: integer(), total_pages: integer(), total: integer(), transactions: [t()]}}
+          | {:error, PluggyElixir.HttpClient.Error.t() | String.t()}
+  def all_by_account(params, config_overrides \\ [])
+
+  def all_by_account(%{account_id: _, from: _, to: _} = params, config_overrides) do
+    @transactions_path
+    |> HttpClient.get(format_params(params), Config.override(config_overrides))
+    |> handle_response
+  end
+
+  def all_by_account(_params, _config_overrides),
+    do: {:error, ":account_id, :from, and :to are required"}
+
+  defp handle_response({:ok, %{status: 200, body: body}}) do
+    result = %{
+      page: body["page"],
+      total_pages: body["totalPages"],
+      total: body["total"],
+      transactions: Enum.map(body["results"], &parse_transaction/1)
+    }
+
+    {:ok, result}
+  end
+
+  defp handle_response({:error, _reason} = error), do: error
+
+  defp format_params(params) do
+    [
+      accountId: params[:account_id],
+      from: params[:from],
+      to: params[:to],
+      pageSize: Map.get(params, :page_size, @default_page_size),
+      page: Map.get(params, :page, @default_page_number)
+    ]
+  end
+
+  defp parse_transaction(transaction) do
```

Comment:
What do you think about creating a module for called `PluggyElixir.Transaction.Parser` that includes this and the functions bellow?

Or even a `PluggyElixir.Transaction.ResponseHandler` that includes the parser functions and the handling ones too.